### ISSUE TITLE
Choose Y-axis when configuring metrics

### ISSUE THREAD

**Issue by 1tylermitchell** [grafana/grafana#2881]
Currently to choose which y-axis your metric will use, you have to configure the metric query, then click on the legend.  This is very unintuitive and hard for new users to find.  When configuring a metric query, a checkbox for "use second y-axis" would be ideal.

Also, when changing metric queries, it often forgets which y-axis you had chosen, so you have to reselect it again (if you remember).  If the settings is available at the metric query design stage then it will remain consistent.

**Comment by torkelo** [comment]
I agree, 

not all TSDBs (especially graphite) make it possible to correlate series returned to which query that caused it to be returned. For graphite (which was the first data source in Grafana) you send a bunch of queries and get a bunch of series back with no way to correlate them. 

Y-axis is defined for a series based on the series name, that is why the setting is lost if you change the series name. You can define y-axis using regex as well. 

But maybe there can be a Y-axis option per query for the TSDBs that do make it possible to correlate this.

**Comment by Dieterbe** [comment]
+1 this would make more sense.

maybe we can contribute a patch to graphite for it to also include the original query in the returned result?  i also had issues with this when working on graph-explorer, i worked around it there by setting a custom alias, but that wouldn't work in grafana because in grafana people need to be able to set their own alias, use aliasByNode, etc.
alternatively, we could execute each query individually first (during graph building time), so we know how the resulting series look like, though if we can get a patch in upstream graphite that would really be ideal IMHO.

**Comment by mtanda** [comment]
+1

**Comment by azrdev** [comment]
Note that this FAQ already mentions this possibility [1], but an option (or at least a pointer) in the metric configuration could still be useful.

[1] <https://github.com/grafana/grafana/wiki/FAQ>

**Comment by alaslums** [comment]
+1 from me on this. Choosing the axes when configuring the metric makes the most sense to me. Clicking on the legend isn't intuitive, it took me 15 mins of googling to find out that's how you get metrics to appear on the other axes.

**Comment by douglasg14b** [comment]
Definitely, took a 1/4 hour to figure this out, and this thread took me to the correct FAQ.

**Comment by bradjorgensen** [comment]
+1

**Comment by PauloTemenos** [comment]
+1 just ran into this. I'm charting 2 different InfluxDB queries, one querying % RAM Available and another querying Bytes RAM Available and they need different axis.

_(manually choosing is not an option in this case, as the data is filtered by a DDL filter that users can change, reverting all results back to left-axis)_

It would be awesome if we could force each of the queries to a specific left or right axis.

**Comment by tcgjensen** [comment]
Any updates on this request - it is a bit of a pain to sit an click on all legends to choose right y axis manually.

**Comment by nagyt234** [comment]
+1 Choosing the axes when configuring the metric makes the most sense also to me. I just created a dashboard with variables. So I had to go the axes definition of all graphs to set special regexp ""Series Specific overrides" as mentioned in the FAQ above (thanks @azrdev). It took me about 30 minutes, ho to find, how to solve the problem.

**Comment by dwgillies** [comment]
+1 to this.  I am still trying to figure out how to select the Y axis in Grafana, I found this page, but even the description at the top is not enough so that I can understand what to do!

**Comment by azrdev** [comment]
@dwgillies you click on the line-colour in the graph legend, there you get a menu with a tab to switch to the right y-axis

![Screenshot_20191024_094843](https://user-images.githubusercontent.com/5132647/67464761-33df6c00-f644-11e9-9396-017765b40f53.png)

**Comment by Dieterbe** [comment]
Torkel, if you decide you want to commit to making this feature, then I'm sure someone of our graphite team can make a patch to allow the correlation between queries and series.

**Comment by dwgillies** [comment]
Thank you.  However, I think it would be so great if there was an option in the dropdown to use Left/Right axes.
![Screen Shot 2019-10-24 at 9 57 00 AM](https://user-images.githubusercontent.com/45215161/67508323-abfa6180-f645-11e9-9c86-2e467d7e650e.png)

**Comment by Dieterbe** [comment]
Spoke with Torkel. He had some good points:
* datasource should return query id, not query used. e.g. you may issue the same query multiple times and set one to left, one to right. you need to differentiate, them.
* torkel thinks the likely implementation would then be display rules that say "if query id was A or C or whatever -> apply these tweaks".

**Comment by douglasg14b** [comment]
Is this still a problem?

**Comment by leeoniya** [comment]
this issue no longer reflects the UI nor the axis configuration capabilities of Grafana in 2025.

the old Graph panel had only two possible y axes, while the new TimeSeries panel has an unlimited amount, and can be placed on the left/right side of your choice. also, y axes are derived automatically from the fields seen in the response data, with extra axes created based on multiple factors, such as field units, min/max settings, and more. here is a list of all properties that when overridden on a field, will create new axes:

https://github.com/grafana/grafana/blob/6787cdccb9670f79ed90a2ea938f7b5a2878c3cd/packages/grafana-ui/src/components/uPlot/internal.ts#L32

it is not possible to represent this flexibility and behavior with a simple toggle. if you want a series on an additional axis you can use field overrides to set a series unit or any of the above properties.
